jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/di/MultipleConstructorInjectTest.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+* Copyright (c) 2021 IBM Corporation.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Ananya Rao
+*******************************************************************************/
+
+package org.eclipse.lsp4jakarta.jdt.di;
+
+import static org.eclipse.lsp4jakarta.jdt.core.JakartaForJavaAssert.*;
+
+import java.util.Arrays;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4jakarta.commons.JakartaDiagnosticsParams;
+import org.eclipse.lsp4jakarta.commons.JakartaJavaCodeActionParams;
+import org.eclipse.lsp4jakarta.jdt.core.BaseJakartaTest;
+import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
+import org.junit.Test;
+
+public class MultipleConstructorInjectTest extends BaseJakartaTest {
+
+    protected static JDTUtils JDT_UTILS = new JDTUtils();
+
+    @Test
+    public void multipleInject() throws Exception {
+        IJavaProject javaProject = loadJavaProject("jakarta-sample", "");
+        IFile javaFile = javaProject.getProject()
+                .getFile(new Path("src/main/java/io/openliberty/sample/jakarta/di/MultipleConstructorWithInject.java"));
+        String uri = javaFile.getLocation().toFile().toURI().toString();
+
+        JakartaDiagnosticsParams diagnosticsParams = new JakartaDiagnosticsParams();
+        diagnosticsParams.setUris(Arrays.asList(uri));
+        
+
+        // test expected diagnostic
+        Diagnostic d1 = d(22, 11, 26,
+                "Inject cannot be used with multiple constructors",
+                DiagnosticSeverity.Error, "jakarta-di", "RemoveInject");
+        
+        Diagnostic d2 = d(27, 11, 26,
+                "Inject cannot be used with multiple constructors",
+                DiagnosticSeverity.Error, "jakarta-di", "RemoveInject");
+        
+        assertJavaDiagnostics(diagnosticsParams, JDT_UTILS, d1,d2);
+       
+        // test expected quick-fix
+        JakartaJavaCodeActionParams codeActionParams1 = createCodeActionParams(uri, d1);
+        TextEdit te = te(21, 1, 22, 4,"");
+        CodeAction ca = ca(uri, "Remove @Inject", d1, te);
+        assertJavaCodeAction(codeActionParams1, JDT_UTILS, ca);
```

Comment:
Is there a quick fix shown for the diagnostics at the second (or more) constructors?

Reply:
Yep, there was a quick fix shown for all constructors. But thanks for the suggestion, I've added a quick fix test for two constructors instead of one.